backend/sql/init.sql

```diff
@@ -0,0 +1,172 @@
+-- All experiments from one experiment data file have the same methodology.
+create table if not exists test_methodology
+(
+    test_methodology_id serial        not null primary key,
+    name                text          not null unique,
+    description         text          null,
+    confluence_link     varchar(1000) null
+);
+
+-- Something that reacts with compounds that scientists want to research.
+-- The goal of their work is to check how a compound affects a target.
+create table if not exists target
+(
+    target_id       serial      not null primary key,
+    name            text        not null unique,
+    creation_date   timestamp   not null
+);
+
+create table if not exists compound
+(
+    compound_id     serial          not null primary key,
+    name            text            not null unique,
+    creation_date   timestamp       not null,
+    chemsoft_link   varchar(1000)   null
+);
+
+-- Corresponds to an Excel file.
+-- Here we store data that is shared by all experiments in one file,
+-- namely methodology and project.
+-- Additionally we store file name and binary contents because
+-- it should be possible to download these files.
+create table if not exists experiment_file
+(
+    experiment_file_id serial not null primary key,
+    project_id         int       not null,
+    methodology_id     int       null,  -- normally not null, but methodology can be removed
+    upload_date        timestamp not null,
+    -- Arbitrary metadata from the file
+    meta               json      null,
+
+    -- Name of the file
+    name               text      not null,
+    -- .xlsx blob
+    contents           bytea     not null,
+
+    constraint belongs_to_project
+        foreign key (project_id) references project (project_id)
+            -- Project id is immutable
+            on update no action
+            -- Deletion is currently not supported for projects
+            on delete no action,
+
+    constraint has_methodology
+        foreign key (methodology_id) references test_methodology (test_methodology_id)
+            -- Methodology id is immutable
+            on update no action
+            -- If the corresponding methodology is removed, we set it to NULL
+            on delete set null
+);
+
+-- Subset of data from a single experiment data file.
+-- It contains a list of experimental measurements for ONE compound and ONE target.
+
+-- You can think about it as a list of concentrations and 1 or more signal values
+-- for each concentration. Usually there are multiple repetitions,
+-- so more than 1 value for each concentration.
+create table if not exists experiment
+(
+    experiment_id       serial  not null primary key,
+    -- File from which this experiment was taken.
+    -- Project, methodology and some other data are inherited from that file.
+    experiment_file_id  int     not null,
+    compound_id         int     not null,
+    target_id           int     not null,
```

Comment:
So, as I can't find smth useful in notion, will take design as a base\
In design we chose `experiment` by project, compound and target, but here I don't see `project_id`. So, if we will add file to this choice we should add `project_id` and corresponding constraints (by the way there is no `has_experiment_file` constraint).

Reply:
`project_id` is present in the `experiment_file` table and we refer to it. So if we want to filter by project, we can filter experiment files first and consider experiments only from those files.

Reply:
But one project may have several experiment types, so with selectors on design we are not able to choose an experiment

Reply:
We don't need to choose one experiment, we need to filter all experiments by projects, compounds and targets. And each of these filters is optional. If someone selects project `proj0`, we will make a request which will fetch all experiments such that experiment file corresponding to this experiment has project equal to `proj0`. I didn't use SQL for a long time, but AFAIK it should be possible and not hard.

Reply:
Please look at dashboard page, we have selecters here, what for?

Reply:
Selectors are filters, there are 3 filters. We show all experiments complying to these filters on the right.

Reply:
Here all 3 filters are used and there is a number of experiments on the right.

![image](https://user-images.githubusercontent.com/2556409/111067543-e502d280-84d5-11eb-83a1-2e25206e5428.png)